pkg/replication/replicator.go

```diff
@@ -244,14 +213,88 @@ func (txr *TxReplicator) disconnect() {
 	txr.logger.Infof("Disconnected from '%s':'%d' for database '%s'", txr.opts.masterAddress, txr.opts.masterPort, txr.db.GetName())
 }
 
-func (txr *TxReplicator) fetchTX() ([]byte, error) {
-	exportTxStream, err := txr.client.ExportTx(txr.clientContext, &schema.ExportTxRequest{Tx: txr.nextTx})
+func (txr *TxReplicator) nextTx() error {
+	commitState, err := txr.db.CurrentState()
 	if err != nil {
-		return nil, err
+		return err
+	}
+
+	syncReplicationEnabled := txr.db.IsSyncReplicationEnabled()
```

Comment:
The syncReplicationEnabled flag is used for both primary and replicas. This limits possible setups - e.g. there may be a replica that is synchronous in relation to its primary but it can also have replicas that are asynchronous. Are we able to handle such case today?

Reply:
yes, that scenario should be handled without issues